src/main/java/nl/tudelft/cse1110/andy/execution/step/RunExternalProcessStep.java

```diff
@@ -0,0 +1,39 @@
+package nl.tudelft.cse1110.andy.execution.step;
+
+import nl.tudelft.cse1110.andy.config.DirectoryConfiguration;
+import nl.tudelft.cse1110.andy.config.RunConfiguration;
+import nl.tudelft.cse1110.andy.execution.Context;
+import nl.tudelft.cse1110.andy.execution.ExecutionStep;
+import nl.tudelft.cse1110.andy.result.ResultBuilder;
+import nl.tudelft.cse1110.andy.execution.process.ExternalProcess;
+
+public class RunExternalProcessStep implements ExecutionStep {
+
+    @Override
+    public void execute(Context ctx, ResultBuilder result) {
+        try {
+            RunConfiguration runConfiguration = ctx.getRunConfiguration();
+            DirectoryConfiguration directoryConfiguration = ctx.getDirectoryConfiguration();
+
+            ExternalProcess externalProcess = runConfiguration.externalProcess();
+
+            if (externalProcess == null) {
+                return;
+            }
+
+            externalProcess.launch();
+
+            ctx.setExternalProcess(externalProcess);
+
+            externalProcess.await();
```

Comment:
Is this call blocking? I can't guess it by the implementation of `await`. If so, this is a problem. We want the rest of the pipeline to work while the other process is executing.

Reply:
This blocks the thread until either the process exits on its own or a predefined string is read from the output (`endSignal`).

Reply:
Oh then this is wrong! Imagine that we are going to run a web app here. This web app must be running when we run the tests. Do we really need this await?

Reply:
This blocks the execution until we read a string from stdout - for instance, we could use this to wait until the web app initialises before running the tests.

Reply:
I'll rename `endSignal` to `initSignal` as it makes more sense that way.